.github/workflows/ci.yml

```diff
@@ -0,0 +1,86 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  schedule:
+    # Run at 18:00 UTC every day
+    - cron: '0 18 * * *'
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # Set 0 to rebase later.
+          fetch-depth: 0
+
+      - name: Read .env
+        run: |
+          . ./.env
+          echo "MDBOOK_VERSION=${MDBOOK_VERSION}" >> $GITHUB_ENV
+          echo "MDBOOK_LINKCHECK_VERSION=${MDBOOK_LINKCHECK_VERSION}" >> $GITHUB_ENV
+          echo "MDBOOK_TOC_VERSION=${MDBOOK_TOC_VERSION}" >> $GITHUB_ENV
+          echo "DEPLOY_DIR=${DEPLOY_DIR}" >> $GITHUB_ENV
+
+      - name: Cache binaries
+        id: mdbook-cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin
+          key: ${{ runner.os }}-${{ env.MDBOOK_VERSION }}--${{ env.MDBOOK_LINKCHECK_VERSION }}--${{ env.MDBOOK_TOC_VERSION }}
+
+      - name: Cache linkcheck
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/book/linkcheck
+          key: ${{ runner.os }}-${{ hashFiles('./book/linkcheck') }}
+
+      - name: Check line lengths
+        if: github.event_name != 'schedule'
+        run: |
+          shopt -s globstar
+          MAX_LINE_LENGTH=100 bash ci/check_line_lengths.sh src/**/*.md
+
+      - name: Install latest nightly Rust toolchain
+        if: steps.mdbook-cache.outputs.cache-hit != 'true'
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: nightly
+            override: true
+
+      - name: Install Dependencies
+        if: steps.mdbook-cache.outputs.cache-hit != 'true'
+        run: |
+          cargo install mdbook --version ${{ env.MDBOOK_VERSION }}
+          cargo install mdbook-linkcheck --version ${{ env.MDBOOK_LINKCHECK_VERSION }}
+          cargo install mdbook-toc --version ${{ env.MDBOOK_TOC_VERSION }}
+
+      - name: Check build
+        if: github.event_name != 'schedule'
+        run: |
+          git switch -c ci
+          git fetch origin master
+          git rebase origin/master
+          git log --oneline | head -n 10
```

Comment:
Why does this rebase over master? Doesn't github do that automatically? What happens on merge conflicts?

Reply:
I just copy-pasted from the travis script, this is no longer required now?

Reply:
(IIRC we introduced it to avoid broken link failures that are already fixed on master.)

Reply:
I know that rust-lang/rust runs CI after rebasing against master; I don't *think* they do that explicitly and github handles it automatically. Not sure though.

Reply:
That sounds surprising to me, but a quick search for 'rebase' in rust's `src/ci` found no result. 🤷‍♂️

Reply:
IIRC this was introduced when we did huge refactorings on librustc, which causes a ton of link failures. Now the broken rate is low, so we could remove rebasing, I think.